src/sidebar/services/annotations.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+const SearchClient = require('../search-client');
+
+// @ngInject
+function annotations(annotationMapper, api, store, streamer, streamFilter) {
+  let searchClient = null;
+
+  /**
+   * Load annotations for all URIs and groupId.
+   *
+   * @param {string[]} uris
+   * @param {string} groupId
+   */
+  function load(uris, groupId) {
+    annotationMapper.unloadAnnotations(store.savedAnnotations());
+
+    // Cancel previously running search client.
+    if (searchClient) {
+      searchClient.cancel();
+    }
+
+    if (uris.length > 0) {
```

Comment:
So here it's waiting for the uri's to load essentially and sometimes when this is called they haven't loaded yet. It's not so much a failure IMO as it is the data is not available to perform a search yet although I suppose it could in theory never be available.

Reply:
@LMS007 @robertknight looking for some feedback/decision here? Do we want to change something or is this ok the way it is?